tests/foreman/ui/test_discoveredhost.py

```diff
@@ -141,6 +135,234 @@ def setUpClass(cls):
         cls.discovery_auto.value = 'True'
         cls.discovery_auto.update({'value'})
 
+        # Create a new Life-Cycle environment
```

Comment:
I like your idea and I completely agree to avoid duplication. 
However, I wanted to keep all provisioning configuration in setup part only and wanted to run that once as there are bunch of tests that required provisioning in place. If I make a helper function  then it needs to be called separately in every test.